src/redirect.rs

```diff
@@ -163,7 +164,12 @@ impl Runnable for InputRedirect {
     fn run(&self) -> Result<String, Box<dyn Error>> {
         let file = File::open(&self.input_file)?;
         let file_fd = file.into_raw_fd();
-        let input = unsafe { Stdio::from_raw_fd(file_fd) };
+
+        let mut child = Command::new("cat")
+            .stdin(unsafe { Stdio::from_raw_fd(file_fd) })
+            .stdout(Stdio::piped())
+            .spawn()?;
+        let input = child.stdout.take().unwrap();
```

Comment:
This (and what I did below) is a bit of a hack :(

Reply:
Sad but needs must :(